packages/components/src/TokenTutorial/TokenTutorialQuiz.tsx

```diff
@@ -0,0 +1,160 @@
+  activeSection: string;
+  skipTutorial: boolean;
+}
+
+export interface TokenTutorialQuizStates {
+  topicIdx: number;
+  activeSection: string;
+  slideIdx: number;
+}
+
+export class TokenTutorialQuiz extends React.Component<TokenTutorialQuizProps, TokenTutorialQuizStates> {
+  public constructor(props: any) {
+    super(props);
+    this.state = { topicIdx: this.props.topicIdx, slideIdx: 0, activeSection: this.props.activeSection };
+  }
+
+  public render(): JSX.Element {
+    const topicIdx = this.state.topicIdx;
+    const slideIdx = this.state.slideIdx;
+    const activeSection = this.state.activeSection;
+
+    switch (activeSection) {
+      case "intro":
+        return (
+          <TutorialContain>
+            <TutorialTopicIntro
+              headerText={TutorialContent[topicIdx].tutorialIntro.header}
+              infoText={TutorialContent[topicIdx].tutorialIntro.content}
+              onClickNext={() => this.next()}
+              onClickSkipTutorial={() => this.skipTutorial()}
+              activeSlide={0}
+              totalSlides={1}
+            />
+          </TutorialContain>
+        );
+      case "tutorial":
+        return (
+          <TutorialContain>
+            <TutorialInfo
+              content={TutorialContent[topicIdx].tutorials[slideIdx].content}
+              onClickPrev={() => this.prev()}
+              onClickNext={() => this.next()}
+              activeSlide={slideIdx + 1}
+              totalSlides={TutorialContent[topicIdx].tutorials.length}
+            />
+          </TutorialContain>
+        );
+      case "quiz":
+        return (
+          <TutorialContain>
+            <TutorialQuestion
+              quizName={TutorialContent[topicIdx].name}
+              quizId={TutorialContent[topicIdx].quizId}
+              question={TutorialContent[topicIdx].questions[slideIdx].question}
+              options={TutorialContent[topicIdx].questions[slideIdx].options}
+              answer={TutorialContent[topicIdx].questions[slideIdx].answer}
+              onClickPrev={() => this.prev()}
+              onClickNext={() => this.next()}
+              activeSlide={slideIdx + 1}
+              totalSlides={TutorialContent[topicIdx].questions.length}
+            />
+          </TutorialContain>
+        );
+      case "completed":
+        return (
+          <TutorialContain>
+            <TutorialTopicCompleted
+              completedHeader={TutorialContent[topicIdx].completed.header}
+              completedText={TutorialContent[topicIdx].completed.content}
+              continueBtnText={TutorialContent[topicIdx].completed.btnText}
+              onClickNextTopic={() => this.nextTopic()}
+            />
+          </TutorialContain>
+        );
+      default:
+        return <></>;
+    }
+  }
+
+  private prev = () => {
+    switch (this.state.activeSection) {
+      case "tutorial":
+        if (this.state.slideIdx === 0) {
+          this.setState({ activeSection: "intro" });
```

Comment:
Might want to combine these `setState` calls, for a few reasons. 

One is every `setState` call will trigger a re-render of any components that are using these -- or components where these states are passed in as props. Another reason is, React performs some optimizations under the hood when calling `setState` and having multiple calls in a row like this may be less predictable / representative of how state will get updated.

Or, if you _really_ want these to be synchronous, you should probably pass arrow functions as the args for these calls.